packages/plugins/typescript/apollo-client-helpers/src/index.ts

```diff
@@ -73,10 +79,16 @@ ${fieldsNames.map(fieldName => `\t${fieldName}?: FieldPolicy<any> | FieldReadFun
 
   return {
     prepend: [
+      config.baseTypesPath
+        ? `import ${config.useTypeImports ? 'type ' : ''}{ Reference } from '@apollo/client';`
+        : null,
       `import ${
         config.useTypeImports ? 'type ' : ''
       }{ FieldPolicy, FieldReadFunction, TypePolicies, TypePolicy } from '@apollo/client/cache';`,
-    ],
+      config.baseTypesPath
+        ? `import ${config.useTypeImports ? 'type ' : ''}* as Types from './${config.baseTypesPath}';`
```

Comment:
Please make sure to provide support for ESM projects by leveraging the `emitLegacyCommonJSImports` option like other plugins do, for example here with the `graphql-request` plugin: 
https://github.com/dotansimha/graphql-code-generator/blob/e743f6984952cbdffa359a3a53eb4c50fe0d2879/packages/plugins/typescript/graphql-request/src/visitor.ts#L44-L47